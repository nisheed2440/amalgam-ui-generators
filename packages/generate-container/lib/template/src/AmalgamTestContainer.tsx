import { observer } from 'mobx-react-lite';
import React, { createContext, useContext } from 'react';
import { IStoreContext, IJsonContext } from './AmalgamTestContainer.interfaces';

// The common store context
export const StoreContext = createContext<IStoreContext>({});
// The JSON data context coming from CMS
export const JsonContext = createContext<IJsonContext>({});

// A memoized component which uses observer
export default observer(() => {
  const store = useContext(StoreContext);
  const json = useContext(JsonContext);
  return (
    <div className="component">
      {json.counterLabel} {store.counter}
    </div>
  );
});
