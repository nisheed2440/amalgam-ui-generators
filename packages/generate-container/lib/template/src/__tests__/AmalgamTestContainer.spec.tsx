import React from 'react';
import TestRenderer from 'react-test-renderer';
import AmalgamTestContainer from '../AmalgamTestContainer';

describe('AmalgamTestContainer', () => {
  it('should render AmalgamTestContainer', () => {
    const AmalgamTestContainerInstance = TestRenderer.create(<AmalgamTestContainer />);
    expect(AmalgamTestContainerInstance).toMatchSnapshot();
  });
});
