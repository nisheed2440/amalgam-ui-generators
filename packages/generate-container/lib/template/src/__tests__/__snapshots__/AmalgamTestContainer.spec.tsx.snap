// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmalgamTestContainer should render AmalgamTestContainer 1`] = `
<div
  className="component"
>
   
</div>
`;
